app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,222 @@
+   * @param args.getMetaMetricsId - A function that returns the current MetaMetrics ID.
+   */
+  constructor({
+    dataDeletionService,
+    messenger,
+    state,
+    getMetaMetricsId,
+  }: {
+    dataDeletionService: PublicInterface<DataDeletionService>;
+    messenger: MetaMetricsDataDeletionControllerMessenger;
+    state?: Partial<MetaMetricsDataDeletionState>;
+    getMetaMetricsId: () => string | null;
+  }) {
+    // Call the constructor of BaseControllerV2
+    super({
+      messenger,
+      metadata,
+      name: controllerName,
+      state: { ...defaultState, ...state },
+    });
+    this.#getMetaMetricsId = getMetaMetricsId;
+    this.#dataDeletionService = dataDeletionService;
+    this.#registerMessageHandlers();
+  }
+
+  /**
+   * Constructor helper for registering this controller's messaging system
+   * actions.
+   */
+  #registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:createMetaMetricsDataDeletionTask`,
+      this.createMetaMetricsDataDeletionTask.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:updateDataDeletionTaskStatus`,
+      this.updateDataDeletionTaskStatus.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:setHasMetaMetricsDataRecorded`,
+      this.setHasMetaMetricsDataRecorded.bind(this),
+    );
+  }
+
+  /**
+   * Creating the delete regulation using source regulation
+   *
+   */
+  async createMetaMetricsDataDeletionTask(): Promise<void> {
+    const metaMetricsId = this.#getMetaMetricsId();
+    if (!metaMetricsId) {
+      throw new Error('MetaMetrics ID not found');
+    }
+
+    const { data } =
+      await this.#dataDeletionService.createDataDeletionRegulationTask(
+        metaMetricsId,
+      );
+    this.update((state) => {
+      state.metaMetricsDataDeletionId = data?.regulateId;
+      state.metaMetricsDataDeletionDate = Date.now();
+      return state;
+    });
+    await this.updateDataDeletionTaskStatus();
+  }
+
+  /**
+   * To check the status of the current delete regulation.
+   */
+  async updateDataDeletionTaskStatus(): Promise<void> {
+    const deleteRegulationId = this.state.metaMetricsDataDeletionId;
+    if (deleteRegulationId.length === 0) {
+      throw new Error('Delete Regulation id not found');
+    }
+
+    const {
+      data: { regulation },
+    } = await this.#dataDeletionService.fetchDeletionRegulationStatus(
+      deleteRegulationId,
+    );
+
+    this.update((state) => {
+      state.metaMetricsDataDeletionStatus = regulation.overallStatus;
+      return state;
+    });
+  }
+
+  /**
+   * To records whether data was collected after last deletion
+   *
+   * @param record - boolean value to determine data is being recorded.
+   */
+  setHasMetaMetricsDataRecorded(record: boolean): void {
+    this.update((state) => {
+      state.hasMetaMetricsDataRecorded = record;
```

Comment:
Why do we need to know when MetaMetrics data has been recorded in the first place?

I'm guessing we wanted to track this because we want to re-enable the "Delete data" button at that point. But... could we instead delete the state about the deletion request? Once metrics are enabled again, it doesn't seem like we use the deletion request data anywhere.